digdag-standards/src/main/java/io/digdag/standards/operator/jdbc/AbstractJdbcResultSet.java

```diff
@@ -0,0 +1,69 @@
+package io.digdag.standards.operator.jdbc;
+
+import java.util.List;
+import java.util.Arrays;
+import java.sql.SQLException;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import com.google.common.collect.ImmutableList;
+
+public abstract class AbstractJdbcResultSet
+    implements JdbcResultSet
+{
+    protected final ResultSet resultSet;
+    private final Object[] results;
+    private final List<Object> list;
+    private final List<String> columnNames;
+
+    public AbstractJdbcResultSet(ResultSet resultSet)
+    {
+        this.resultSet = resultSet;
+        try {
+            ResultSetMetaData meta = resultSet.getMetaData();
+            int columnCount = meta.getColumnCount();
+            this.results = new Object[columnCount];
+            this.list = Arrays.asList(results);
+
+            ImmutableList.Builder<String> names = ImmutableList.builder();
+            for (int i=0; i < columnCount; i++) {
+                names.add(meta.getColumnLabel(i + 1));  // JDBC column index begins from 1
+            }
+            this.columnNames = names.build();
+        }
+        catch (SQLException ex) {
+            throw new DatabaseException("Failed to fetch result metadata", ex);
+        }
+    }
+
+    @Override
+    public List<String> getColumnNames()
+    {
+        return columnNames;
+    }
+
+    @Override
+    public List<Object> next()
+    {
+        try {
+            if (!resultSet.next()) {
+                return null;
+            }
+            getObjects(results);
```

Comment:
The return value `Arrays.asList(results)` shares the instance variable`Object[] results` internally. We'd better instantiate a new `Object[]` here for a return value.